textattack/transformations/word_swap_change_number.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+from word2number import w2n
+
+from textattack.transformations import Transformation
+
+
+def idx_to_words(ls, words):
+    """Given a list generated from cluster_idx, return a list that contains
+    sub-list (the first element being the idx, and the second element being the
+    words corresponding to the idx)"""
+
+    output = []
+    for cluster in ls:
+        word = words[cluster[0]]
+        for idx in cluster[1:]:
+            word = " ".join([word, words[idx]])
+        output.append([cluster, word])
+    return output
+
+
+def cluster_idx(idx_ls):
+    """Given a list of idx, return a list that contains sub-lists of adjacent
+    idx."""
+
+    if len(idx_ls) < 2:
+        return [[i] for i in idx_ls]
+    else:
+        output = [[idx_ls[0]]]
+        prev = idx_ls[0]
+        list_pos = 0
+
+        for idx in idx_ls[1:]:
+            if idx - 1 == prev:
+                output[list_pos].append(idx)
+            else:
+                output.append([idx])
+                list_pos += 1
+            prev = idx
+        return output
+
+
+class WordSwapChangeNumber(Transformation):
+    def __init__(self, max_change=1, n=3, **kwargs):
+        """A transformation that recognize numbers in sentence, and return
+        sentences with altered numbers.
+
+        :param max_change: Maximum percent of change (1 being 100%)
+        :param n: Numbers of new numbers to generate
+        """
+        super().__init__(**kwargs)
+        self.max_change = max_change
+        self.n = n
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        words = current_text.words
+        num_idx = []
+        num_words = []
+
+        # find indexes of alphabetical words
+        for idx in indices_to_modify:
+            word = words[idx].lower()
+            for number in STR_NUM:
+                if number in word:
+                    if word in ["point", "and"]:
+                        if 0 < idx and (idx - 1) in num_idx:
+                            num_idx.append(idx)
+                    else:
+                        num_idx.append(idx)
+                    break
+
+            if word.isdigit():
+                num_words.append([[idx], word])
+
+        # cluster adjacent indexes to get whole number
+        num_idx = cluster_idx(num_idx)
+        num_words += idx_to_words(num_idx, words)
+
+        # replace original numbers with new numbers
+        transformed_texts = []
+        for num_word in num_words:
+            idx = num_word[0]
+            word = num_word[1]
+            replacement_words = self._get_new_number(word)
+            for r in replacement_words:
+                if r == word:
+                    continue
+                text = current_text.replace_word_at_index(idx[0], str(r))
+                if len(idx) > 1:
+                    index = idx[1]
+                    for i in idx[1:]:
+                        text = text.delete_word_at_index(index)
```

Comment:
I've seen this pattern in a couple places: to replace multiple words with a single word, you replace the first word, then delete the rest. That's a good solution. However, I think you can save some code by using the `replace_words_at_indices` method from AttackedText (in all places where you do this, including here!).

In this case, I think it would look like:
```text = current_text.replace_words_at_indices(idx, str(r))```

Reply:
The problem with using `replace_words_at_indices` is that we can only replace n idx with exactly n words, or it's going to give this error: ` f"Cannot replace {len(new_words)} words at {len(indices)} indices."`

Should we change `replace_words_at_indices ` to accompany that?

Reply:
go for it 👍